qa/vagrant/src/main/java/org/elasticsearch/packaging/test/WindowsServiceTestCase.java

```diff
@@ -0,0 +1,209 @@
+        //
+        // this won't persist to another session so we don't have to reset anything
+        return sh.runIgnoreExitCode(
+            "$Env:PATH = '" + newPath + "'; " +
+                "Remove-Item Env:JAVA_HOME; " +
+                script
+        );
+    }
+
+    public void test10InstallArchive() {
+        installation = installArchive(distribution());
+        verifyArchiveInstallation(installation, distribution());
+        serviceScript = installation.bin("elasticsearch-service.bat").toString();
+    }
+
+    public void test11InstallServiceExeMissing() throws IOException {
+        Path serviceExe = installation.bin("elasticsearch-service-x64.exe");
+        Path tmpServiceExe = serviceExe.getParent().resolve(serviceExe.getFileName() + ".tmp");
+        Files.move(serviceExe, tmpServiceExe);
+        Result result =  sh.runIgnoreExitCode(serviceScript + " install");
+        assertThat(result.exitCode, equalTo(1));
+        assertThat(result.stdout, containsString("elasticsearch-service-x64.exe was not found..."));
+        Files.move(tmpServiceExe, serviceExe);
+    }
+
+    public void test12InstallService() {
+        sh.run(serviceScript + " install");
+        assertService(DEFAULT_ID, "Stopped", DEFAULT_DISPLAY_NAME);
+        sh.run(serviceScript + " remove");
+    }
+
+    public void test13InstallMissingJava() throws IOException {
+        Result result = runWithoutJava(serviceScript + " install");
+        assertThat(result.exitCode, equalTo(1));
+        assertThat(result.stderr, containsString("could not find java; set JAVA_HOME or ensure java is in PATH"));
+    }
+
+    public void test14RemoveNotInstalled() {
+        Result result = sh.runIgnoreExitCode(serviceScript + " remove");
+        assertThat(result.stdout, result.exitCode, equalTo(1));
+        assertThat(result.stdout, containsString("Failed removing '" + DEFAULT_ID + "' service"));
+    }
+
+    public void test20CustomizeServiceId() {
+        String serviceId = "my-es-service";
+        String displayName = DEFAULT_DISPLAY_NAME.replace(DEFAULT_ID, serviceId);
+        sh.getEnv().put("SERVICE_ID", serviceId);
+        sh.run(serviceScript + " install");
+        assertService(serviceId, "Stopped", displayName);
+        sh.run(serviceScript + " remove");
+    }
+
+    public void test21CustomizeServiceDisplayName() {
+        String displayName = "my es service display name";
+        sh.getEnv().put("SERVICE_DISPLAY_NAME", displayName);
+        sh.run(serviceScript + " install");
+        assertService(DEFAULT_ID, "Stopped", displayName);
+        sh.run(serviceScript + " remove");
+    }
+
+    // NOTE: service description is not attainable through any powershell api, so checking it is not possible...
+
+    public void test30StartStop() throws IOException {
+        sh.run(serviceScript + " install");
+        sh.run(serviceScript + " start");
+        ServerUtils.waitForElasticsearch();
```

Comment:
Usually we also make a few API calls to the server, e.g. https://github.com/elastic/elasticsearch/blob/2aba52de8f9315b0e384e1c657d7b0401d26a1b0/qa/vagrant/src/main/java/org/elasticsearch/packaging/test/PackageTestCase.java#L121-L122

I'm not completely sold on the value of those though

Reply:
Sure, I've added calling the same test method here.